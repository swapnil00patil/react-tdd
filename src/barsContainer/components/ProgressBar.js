import React from 'react'
import styled from 'styled-components'

const Bar = styled.div`
  height: 38px;
  width: ${props => props.barValue}%;
  background: #b3d7e6;
  margin-bottom: 10px;
`
const Wrap = styled.div`
  height: 38px;
  width: 100%;
  background: transparent;
  margin-bottom: 10px;
  border: 1px solid #000;
`

class ProgressBar extends React.PureComponent {
  render () {
    const { barValue } = this.props
    return (
      <Wrap>
        <Bar barValue={barValue} />
      </Wrap>
    )
  }
}

export default ProgressBar
