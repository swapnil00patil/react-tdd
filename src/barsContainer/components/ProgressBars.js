import React from 'react'

import ProgressBar from './ProgressBar'

class ProgressBars extends React.PureComponent {
  render () {
    const { bars } = this.props
    return bars.map((value, index) => <ProgressBar key={index} barValue={value} />)
  }
}

export default ProgressBars
