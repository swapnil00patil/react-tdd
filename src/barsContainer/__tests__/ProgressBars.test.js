import React from 'react'
import { shallow, mount } from 'enzyme'

import ProgressBars from '../components/ProgressBars'
import ProgressBar from '../components/ProgressBar'

describe('ProgressBars', () => {
  it('ProgressBars should be defined', () => {
    expect(ProgressBars).toBeDefined()
  })

  it('should render exactly same number of ProgressBars', () => {
    const wrapper = mount(<ProgressBars bars={[12, 13, 14]} />)
    expect(wrapper.find(ProgressBar).length).toBe(3)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<ProgressBars bars={[12, 13, 14]} />)
    expect(wrapper).toMatchSnapshot()
  })
})
