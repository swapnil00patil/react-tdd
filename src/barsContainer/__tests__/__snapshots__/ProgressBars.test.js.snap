// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBars should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBars
    bars={
        Array [
            12,
            13,
            14,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <ProgressBar
      barValue={12}
/>,
    <ProgressBar
      barValue={13}
/>,
    <ProgressBar
      barValue={14}
/>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <ProgressBar
        barValue={12}
/>,
      <ProgressBar
        barValue={13}
/>,
      <ProgressBar
        barValue={14}
/>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
